universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java

```diff
@@ -0,0 +1,117 @@
+package views.admin.questions;
+
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.br;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.label;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Optional;
+import play.mvc.Http.Request;
+import play.twirl.api.Content;
+import services.question.QuestionDefinition;
+import services.question.QuestionType;
+import services.question.TranslationNotFoundException;
+import views.BaseHtmlLayout;
+import views.BaseHtmlView;
+
+public final class QuestionEditView extends BaseHtmlView {
+  private final BaseHtmlLayout layout;
+
+  @Inject
+  public QuestionEditView(BaseHtmlLayout layout) {
+    this.layout = layout;
+  }
+
+  public Content render(Request request, Optional<QuestionDefinition> question) {
+    String headerText = question.isPresent() ? "Edit Question" : "New Question";
+    return layout.htmlContent(
+        body(
+            renderHeader(headerText),
+            buildQuestionForm(question).with(makeCsrfTokenInputTag(request))));
+  }
+
+  private ContainerTag buildQuestionForm(Optional<QuestionDefinition> optionalDefinition) {
+    String buttonText = "";
+
+    ContainerTag formTag = form().withMethod("POST");
+
+    Optional<String> questionText = Optional.empty();
+    Optional<String> questionHelpText = Optional.empty();
+
+    if (optionalDefinition.isPresent()) { // Editing a question.
+      QuestionDefinition definition = optionalDefinition.get();
+      buttonText = "Update";
+      formTag.withAction(
+          controllers.admin.routes.QuestionController.edit(definition.getPath()).toString());
```

Comment:
question: did you mean `update`? Also update takes an `id`, maybe should be `path`.

Reply:
Yup. It's getting late.